docs/cancellation.md

```diff
@@ -0,0 +1,387 @@
+# Cancellation
+Sometimes, requests take a long time to service and clients disconnect
+before Synapse produces a response. To avoid wasting resources, Synapse
+can cancel request processing for select endpoints with the `@cancelled`
```

Comment:
```suggestion
can cancel request processing for select endpoints with the `@cancellable`
```

Reply:
Good spot, let's fix that.

I'll write some more words on the decorator.